lib/Conversion/TorchToTMTensor/TorchToTMTensor.cpp

```diff
@@ -1600,27 +1600,77 @@ class ConvertAtenScaledDotProductAttentionOp
                                            "only default scale supported");
     }
 
+    auto opTy = cast<ValueTensorType>(op.getType()).toBuiltinTensor();
+    auto query = adaptor.getQuery();
+    auto value = adaptor.getValue();
+    auto key = adaptor.getKey();
+
+    auto loc = op.getLoc();
+    auto collapseBatch = [&rewriter, loc](Value value) -> Value {
+      auto valueTy = cast<ShapedType>(value.getType());
+      if (valueTy.getRank() <= 3)
```

Comment:
If the rank is less than 3, should we emit a failure because we can't codegen that?

Reply:
Notify match error added.